Lib/urllib/parse.py

```diff
@@ -865,7 +865,7 @@ def splithost(url):
     """splithost('//host[:port]/path') --> 'host[:port]', '/path'."""
     global _hostprog
     if _hostprog is None:
-        _hostprog = re.compile('^//([^/?]*)(.*)$')
+        _hostprog = re.compile('//([^/#?]*)(.*)', re.DOTALL)
```

Comment:
I'm not sure about the usage of the DOTALL flag, see:
http://bugs.python.org/issue30500#msg296422